File: src/main/java/com/example/ddalkkak/db/repository/LinkInfoRepository.java
package com.example.ddalkkak.db.repository;

import com.example.ddalkkak.db.model.LinkInfo;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;

public interface LinkInfoRepository extends JpaRepository<LinkInfo, Long> {
    List<LinkInfo> findByUser(String user);

    List<LinkInfo> findAllByOrderBySlackCreatedAtDesc();

    List<LinkInfo> findBySlackCreatedAt(LocalDate slackCreatedAt);

}

File: DdalkkakAndroid/app/src/main/java/com/example/ddalkkak_android/repository/LinkInfoRepository.kt
package com.example.ddalkkak_android.repository

import com.example.ddalkkak_android.data.LinkInfo

interface LinkInfoRepository {
    suspend fun getLinkInfo(): List<LinkInfo>
    suspend fun getLinkInfoByUser(user: String): List<LinkInfo>
    suspend fun getLinkInfoByCreated(created: String): List<LinkInfo>
    suspend fun getCreatedAts(): List<String>
    suspend fun getSearch(keyword: String): List<LinkInfo>
}